hc/blog/views.py

```diff
@@ -0,0 +1,55 @@
+from django.shortcuts import render
+from django.utils import timezone
+from django.shortcuts import render, get_object_or_404
+from django.shortcuts import redirect
+from hc.blog.forms import PostForm
+from hc.blog.models import Post, Category
+
+
+def post_list(request):
+    # Retrieve all posts from database and save to dictionary
+    posts = Post.objects.filter(published_date__lte=timezone.now()).order_by('published_date')
+    all_posts = []
+    for article in posts:
+        one_article = {
+            "title":article.title,
+            "text":article.text[:100]+"...",
```

Comment:
There is a template helper [truncate_chars](https://docs.djangoproject.com/en/1.10/ref/templates/builtins/#truncatechars)